version-published/checkbox_version_published.py

```diff
@@ -0,0 +1,346 @@
+    # into the track and the risk
+
+    def matches_spec(channel_info: dict) -> bool:
+        # some of the channel names don't follow the "track/risk" format
+        # so we need to handle them separately
+        if "/" not in snap_spec.channel:
+            return (
+                channel_info["channel"]["name"] == snap_spec.channel
+                and channel_info["version"] == snap_spec.version
+                and channel_info["channel"]["architecture"] in snap_spec.arch
+            )
+        else:
+            track, risk = snap_spec.channel.split("/")
+            return (
+                channel_info["channel"]["track"] == track
+                and channel_info["channel"]["risk"] == risk
+                and channel_info["version"] == snap_spec.version
+                and channel_info["channel"]["architecture"] in snap_spec.arch
+            )
+
+    return any(
+        matches_spec(channel_info)
+        for channel_info in store_response["channel-map"]
+    )
+
+
+def check_snaps_availability(
+    snap_specs: list[SnapSpec],
+    snaps_available: dict[SnapSpec, bool],
+) -> None:
+    print("Checking if the snaps are available ...")
+    # Record of snaps for which we've already fetched the data from the store.
+    for snap_spec in snap_specs:
+        # Only fetch from the store if not already fetched and not already
+        # available.
+        if snaps_available[snap_spec]:
+            continue
+        try:
+            store_response = get_snap_info_from_store(snap_spec)
+            snaps_available[snap_spec] = is_snap_available(
+                snap_spec, store_response
+            )
+
+        except (requests.RequestException, RuntimeError) as exc:
+            # Handle request exceptions but continue the loop.
+            print(f"Error while querying the snap store: {exc}")
+
+    # Print the list of snaps that were not found.
+    not_available = [
+        snap_spec
+        for snap_spec, is_available in snaps_available.items()
+        if not is_available
+    ]
+    if not_available:
+        print("Not all snaps were available in the store.")
+        print("Here is the list of snaps that were not found:")
+        print("name       | channel     | arch ")
+        print("-" * 32)
+        for snap_spec in not_available:
+            print(
+                f"{snap_spec.name:<10} | {snap_spec.channel:<11} | "
+                f"{snap_spec.arch:<5}"
+            )
+        print("-" * 32 + "\n")
+    else:
+        print("All snaps were found.")
+
+
+def get_package_specs(yaml_content: dict, version: str) -> list[PackageSpec]:
+    """
+    Create a list of PackageSpec objects from the yaml content.
+    :param yaml_content: file containing the package requirements
+    :param version: the version of the package
+    :return: the list of PackageSpec objects
+    """
+    version = version.replace("-", "~")
+    if "required-packages" not in yaml_content:
```

Comment:
we expect the yaml to be structured as a dict, so if we actually codify it (see below for details) then we can streamline the whole thing here to just do:

`yaml_content.get("required_packages", [])`
This way no need for branching.

Same goes for snaps ofc.